src/integration-tests/src/storage.rs

```diff
@@ -235,8 +237,19 @@ async fn folders(client: &storage_control::client::Storage, bucket_name: &str) -
         "missing folder name {folder_name} in {folder_names:?}"
     );
 
+    println!("\nTesting rename_folder()");
+    let rename = client
```

Comment:
I did not think about this before: can you delete a bucket that has folders in it? Even if the folders are empty?  If we cannot, then the `cleanup_stale_buckets()` function needs some fixes.

Reply:
> can you delete a bucket that has folders in it?

Yes, I commented out the `delete_folder()` call in the integration test, and the subsequent `delete_bucket()` call still succeeded.